src/index.js

```diff
@@ -29,70 +29,91 @@ function App() {
+    },[todos]); // this effect only fires if the todos array changes: which is toggle complete and remove tasks
 
-    function handleAddItem(e) {
+    const handleAddItem = () => {
         const currentInput =  toDoInput.current.value;
         const pickedDueDate = dueDateInput.current.value;
+        const currentTodos = [...todos];
 
-        if(currentInput && pickedDueDate) {
-            setToDos(prevList => {
-                return [...prevList, { id:uuidv4(), name: currentInput, due: pickedDueDate, completed: false}]
-            })
+        if(currentInput && pickedDueDate){ // will implement validations in next step
+            setToDos([...currentTodos, { id:uuidv4(), name: currentInput, due: pickedDueDate, completed: false}]);
         }
         toDoInput.current.value = null;
         dueDateInput.current.value = null;
     }
 
-    function toggleCompleted(e) {
+        const toggleCompleted = (e) => {
         const currentTodos = [...todos];
         const toToggleItem = currentTodos.find(eachCurrentTodo => eachCurrentTodo.id === e);
         toToggleItem.completed = !toToggleItem.completed;
         setToDos(currentTodos);
     }
 
-    function handleClearItems() {
+    const handleClearItems = () => {
         const currentTodos = [...todos];
         const toRemainItems = currentTodos.filter(eachCurrentTodo => !eachCurrentTodo.completed);
         setToDos(toRemainItems);
     }
 
-    function sortBy(e) {
-        const colToSort = e.target.value;
-        const currentSort = sortConfig;
-        const currentTodos = [...todos];
+    const toggleSort = (e) => { // we only edit the sortConfig here, nothing else
+        const colToSort = e;
         let direction;
+        let newSortConfig = {};
 
-        if(colToSort){
-            if(!currentSort[colToSort]){
-                direction = 'asc';
-            }
-            else {
-                direction = currentSort[colToSort] === 'asc'? 'desc' : 'asc'; 
-            }
-            currentSort[colToSort] = direction;
+        // goal is to ony 1 key-value pair at any time, so the render action have only 1 thing to recognize even without a click trigger
+        // 1. if exists -->> flip it
+        // 2. if dont exists --> destroy everything in it, create this key
+        if(sortConfig[colToSort]){
+            direction = sortConfig[colToSort] === 'asc'? 'des' : 'asc';
+        }
+        else{
+            direction = 'asc';
+        }
+        newSortConfig[colToSort] = direction;
+        setSortConfig(newSortConfig);
+    }
 
-            setSortConfig(currentSort);
+    const renderList = () => { // or this can just be a simple render function, that checks for sort btw every time
+        const currentTodos = [...todos];
+        // not sure if this is a good practice t o use Object.entries here, but now there is always only 1 pair of key:value, this should be good functionally?
+        const currentSortConfig = Object.entries(sortConfig);
```

Comment:
You're throwing away the advantage of storing as object. If you want to convert to array then you should just store it as array, however I do not recommend array.

What you can do is store your `sortConfig` as 
```js
{
  columnKey: 'date',
  direction: 'asc',
}
```
And now you dont have to convert it to array.